test/integration/annotation/annotated_resource_test.go

```diff
@@ -0,0 +1,190 @@
+		},
+		{
+			name:          "services-annotated",
+			doc:           "Services are annotated after Skupper is deployed and exposed service list matches",
+			modification:  addAnnotation,
+			expectedSites: 2,
+			expectedServicesProto: map[string]string{
+				"nginx-1-dep-web":          "tcp",
+				"nginx-2-dep-web":          "tcp",
+				"nginx-1-svc-exp-notarget": "tcp",
+				"nginx-2-svc-exp-notarget": "tcp",
+				"nginx-1-svc-exp-target":   "http",
+				"nginx-2-svc-exp-target":   "http",
+			},
+		},
+	}
+
+	// Test flow
+	defer TearDown(t)
+	base.HandleInterruptSignal(t, TearDown)
+
+	// 1. Initialize Namespaces
+	Setup(t)
+
+	// 2. Deploying the Pre-annotated resources
+	DeployResources(t)
+
+	// 3. Deploys Skupper and create a VAN between both clusters
+	CreateVan(t)
+
+	// 4. Iterate through test table and run each test
+	for _, test := range testTable {
+		t.Run(test.name, func(t *testing.T) {
+			var err error
+
+			// 4.1. If test expects modifications to be performed, run them
+			if test.modification != nil {
+				test.modification(t)
+			}
+
+			// 4.2 Validate services from DATA endpoint (with retries)
+			var resp *tools.CurlResponse
+			var consoleData ConsoleData
+			err = utils.Retry(interval, 20, func() (bool, error) {
+				resp, err = tools.Curl(cluster1.KubeClient, cluster1.RestConfig, cluster1.Namespace, "", dataEndpoint, curlOpts)
+				if err != nil {
+					return false, nil
+				}
+
+				// 4.2.1. Parsing ConsoleData response
+				if err = json.Unmarshal([]byte(resp.Body), &consoleData); err != nil {
+					return false, nil
+				}
+
+				// 4.2.2. retry till expected amount of sites and services match
+				if test.expectedSites != len(consoleData.Sites) || len(test.expectedServicesProto) != len(consoleData.Services) {
+					return false, nil
+				}
+
+				// 4.2.3. populating map of services/protocols returned
+				servicesFound := map[string]string{}
+				for _, svc := range consoleData.Services {
+					svcMap, ok := svc.(map[string]interface{})
+					assert.Assert(t, ok)
+					address, ok := svcMap["address"].(string)
+					assert.Assert(t, ok)
+					protocol, ok := svcMap["protocol"].(string)
+					assert.Assert(t, ok)
+					servicesFound[address] = protocol
+				}
+
+				// 4.2.4. only consider as done if all match (otherwise it will timeout)
+				res := reflect.DeepEqual(test.expectedServicesProto, servicesFound)
+				return res, nil
+			})
+			assert.Assert(t, err, "timed out waiting for expected services/protocol list to match")
+
+			// 4.3. Validating all exposed services are reachable across clusters
+			for _, cluster := range []*vanClient.VanClient{cluster1, cluster2} {
+				for svc := range test.expectedServicesProto {
+					t.Logf("validating communication with service %s through %s", svc, cluster.Namespace)
+					// reaching service through proxy-controller's pod (with some attempts to make sure bridge is connected)
+					err = utils.Retry(interval, 20, func() (bool, error) {
```

Comment:
another hardcode 20. sorry for repeiting...

Reply:
haha, that is fine.